File: downloadablebookandexercisecode/1256/code_files/Chapter_01_USACO_Bronze/USACO_2019_feb_b1_herding.cpp
// http://usaco.org/index.php?page=viewproblem2&cpid=915

#include <iostream>

using namespace std;

int main() {
    freopen("herding.in", "r", stdin);
    freopen("herding.out", "w", stdout);

    int a, b, c;
    cin >> a >> b >> c;

    int ans_min = 0;
    int ans_max = 0;

    // arrange a,b,c in order
    // poor men's bubble sort. pushing the smallest to a

    int tmp;    // We can use swap(arg1, arg2) instead
    if (c < b) {
        tmp = b;
        b = c;
        c = tmp;
    }
    if (b < a) {
        tmp = a;
        a = b;
        b = tmp;
    }  // by here a is smallest
    if (c < b) {
        tmp = b;
        b = c;
        c = tmp;
    }

    // Algorithm
    ans_min = 2; // we can always do it in 2
    if (c - a == 2) {
        ans_min = 0;
    } else {
        if (c - b == 2 || b - a == 2) {
            ans_min = 1;
        } else {
            ans_min = 2;
        }
    }

    ans_max = max(c - b - 1, b - a - 1);

    cout << ans_min << endl << ans_max << endl;

}

File: downloadablebookandexercisecode/1256/code_files/Chapter_01_USACO_Bronze/USACO_2019_feb_b1_herding.py
# http://usaco.org/index.php?page=viewproblem2&cpid=915

f_in = open('herding.in', 'r')

a, b, c = map(int, f_in.readline().split())

# sort the 3 cows
# there are many more concise ways to do it in python.
# we'll just use the most verbose one.
if c < b:
    # b, c = c, b  # swap the two values
    tmp = b
    b = c
    c = tmp
if b < a:
    a, b = b, a

ans_min = 2
if c - a == 2:
    ans_min = 0
else:
    if c - b == 2 or b - a == 2:
        ans_min = 1
    else:
        ans_min = 2

ans_max = max(c - b - 1, b - a - 1)

f_out = open('herding.out', 'w')
f_out.write(f'{ans_min}\n{ans_max}\n')
f_out.close()
